### PR TITLE
fix(ilp): multiple bugs, see the description

### LINKED ISSUE #1213: ILP does not handle duplicate fields in the message
**Describe the bug**
ILP messes data if message has duplicate filed tags

**To Reproduce**
Send
```
weather alpha=0.1,speed=1,speed=2
weather alpha=0.2
weather timestamp=9831289
```

This will result in messed data at line 2 having `speed` = 2 and line 3 `timestamp` in 1970

**Expected behavior**
Reject line or take either first or last value of duplicated fileds. Reject lines where designated timestamp is specified as a field

**Screenshots**

**Environment (please complete the following information):**
 - Version: [e.g. 5.0.4]

**Additional context**
related to #1212 

Can be reproduced with following test in LineTcpServerTest.java 

```Java
    @Test
    public void testDubFieldValues() throws Exception {
        String lineData =
                "weather g=1,a=0.1,windspeed=1.0 0\n" +
                "weather g=2,windspeed=2.0,windspeed=3.0 60000000000\n" +
                "weather g=3,a=0.4,a=0.5 120000000000\n"+
                "weather g=4,a=0.7,windspeed=67.72 180000000000\n";

        runInContext(() -> {
            send(lineData, "weather", true, false);

            String expected =
                    "g\ta\twindspeed\ttimestamp\n" +
                            "1.0\t0.1\t1.0\t1970-01-01T00:00:00.000000Z\n" +
                            "2.0\tNaN\t2.0\t1970-01-01T00:01:00.000000Z\n" +
                            "3.0\t0.4\tNaN\t1970-01-01T00:02:00.000000Z\n" +
                            "4.0\t0.67\t67.72\t1970-01-01T0:03:00.000000Z\n";
            assertTable(expected, "weather");
        });
    }

```

### LINKED ISSUE #1608: Incorrect line protocol usage causes SEGFAULT or corruption of timestamps every other row
### Describe the bug

Consider a table that declares one of its columns as a designated timestamp.

After reading the [ILP guide](https://questdb.io/docs/guides/influxdb-line-protocol/), it's not obvious how to write valid ILP messages to insert data into the table:

- (a) include the designated timestamp as a key-value pair, omiting the ILP timestamp?
- (b) include the designated timestamp as the ILP timestamp, omiting the key-value pair?
- (c) include the designated timestamp twice, once for the ILP timestamp and one for the key-value pair?

On QuestDB 6.1.1 the correct answer seems to be (b). Unfortunately the other two don't generate errors but cause either silent timestamp corruption or a crash.

- (a) strangely causes _every other_ designated timestamp (on alternating rows) to be set to the current time instead of the given value, a bit like the issue reported in #548
- (c) works the first time, but then causes the crash reported in #1338

### To reproduce

```
CREATE TABLE test(time_close TIMESTAMP, number LONG) timestamp(time_close)
```

Option (a):
```
test, number=1i,time_close=1466111018337510t\n
test, number=2i,time_close=1466111018337520t\n
test, number=3i,time_close=1466111018337530t\n
test, number=4i,time_close=1466111018337540t\n
```
Option (b):
```
test, number=1i 1466111018337510000\n
test, number=2i 1466111018337520000\n
test, number=3i 1466111018337530000\n
test, number=4i 1466111018337540000\n
```
Option (c):
```
test, number=1i,time_close=1466111018337510 1466111018337510000\n
test, number=2i,time_close=1466111018337520 1466111018337520000\n
test, number=3i,time_close=1466111018337530 1466111018337530000\n
test, number=4i,time_close=1466111018337540 1466111018337540000\n
```

### Expected Behavior

Ideally the guide should cover the case of designated timestamps.
QuestDB should also throw an error on ingesting these malformed ILP messages or accept them without crashing or corruption.

### Environment

```markdown
- **QuestDB version**: 6.1.1
- **OS**: MacOS 12.0.1
```

### Additional context

_No response_

### LINKED ISSUE #1618: Potential race in ILP ingestion and DDL queries
### Describe the bug

When ILP data ingestion starts the measurements (rows) are being parsed and stored along with column indexes (`LineTcpMeasurementEvent#createMeasurementEvent()`). The writer (think, the writer lock) is being acquired by the writer thread later in async fashion (`LineTcpMeasurementEvent#processMeasurementEvent()`). Event processing code uses column indexes previously stored when creating the measurement event.

There is an edge case here. If a DDL gets executed concurrently with ILP data ingestion, it may be able to drop and/or columns which indexes are stored in the in-flight measurement events. This may leads to various issues, like NPE.

### To reproduce

Here a simple way to reproduce the race manually with some code modification. Reproducing it in tests more or less reliably should simple enough.

1. Add `Os.sleep(60_000);` at the beginning of the `processMeasurementEvent()` method.
2. Start the server and execute the following DDL in Web Console:
```sql
CREATE TABLE trades (ts TIMESTAMP, name SYMBOL capacity 256 CACHE, value DOUBLE) timestamp (ts) PARTITION BY DAY;
```
3. Run one of ILP ingestion scripts from https://questdb.io/docs/develop/insert-data/#influxdb-line-protocol
4. Execute the following DDL in Web Console:
```sql
ALTER TABLE trades DROP COLUMN value;
```
5. Wait ~1 minute and check server logs:
```
2021-11-26T06:38:54.706342Z I i.q.c.p.WriterPool >> [table=`trades`, thread=14]
2021-11-26T06:38:54.706899Z I i.q.c.TableWriter switched partition [path='/tmp/questdb/db/trades/2021-11-26]
2021-11-26T06:38:54.707537Z E i.q.c.l.t.LineTcpMeasurementScheduler closing writer for because of error [table=trades,ex=
java.lang.NullPointerException
	at io.questdb.cairo.BaseRecordMetadata.getColumnType(BaseRecordMetadata.java:60)
	at io.questdb.cutlass.line.tcp.LineTcpMeasurementScheduler$LineTcpMeasurementEvent.processMeasurementEvent(LineTcpMeasurementScheduler.java:821)
	at io.questdb.cutlass.line.tcp.LineTcpMeasurementScheduler$WriterJob.drainQueue(LineTcpMeasurementScheduler.java:1331)
	at io.questdb.cutlass.line.tcp.LineTcpMeasurementScheduler$WriterJob.run(LineTcpMeasurementScheduler.java:1277)
	at io.questdb.mp.Worker.run(Worker.java:111)
]
```

### Expected Behavior

The measurement gets rejected with proper error handling or is processed correctly.

### Environment

```markdown
- **QuestDB version**: latest 6.1.3-SNAPSHOT
- **OS**: Ubuntu 20.04
- **Browser**:
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by bluestreak01**
This a refactoring PR of https://github.com/questdb/questdb/pull/1658, please review that PR first for the functional changes. This particular PR are non-functional changes for purpose of improving code flow and code maintenance.

I'm copying description of what is fixed here:

closes #1213, closes #1608, closes #1618

https://github.com/questdb/questdb/issues/1213 - ILP does not handle duplicate fields in the message
https://github.com/questdb/questdb/issues/1608 - Incorrect line protocol usage causes SEGFAULT or corruption of timestamps every other row
https://github.com/questdb/questdb/issues/1618 - Potential race in ILP ingestion and DDL queries

Also fixed yet undiscovered bugs caused by mixing UTF8 and UTF16 keys in the same map instances.

Will update the doc to explain how duplicate fields are handled.
https://questdb.io/docs/guides/influxdb-line-protocol

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 897 / 923 (97.18%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/DefaultColumnTypes.java|14|15|93.33%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpWriterJob.java|92|95|96.84%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementEvent.java|363|375|96.80%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java|133|138|96.38%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableStructureAdapter.java|41|42|97.62%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableUpdateDetails.java|92|94|97.87%|
|:large_blue_circle:|io/questdb/std/BoolList.java|91|92|98.91%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpNetworkIOJob.java|51|52|98.08%|
|:large_blue_circle:|io/questdb/std/IntList.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|17|17|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableUtils.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 897 / 923 (97.18%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/DefaultColumnTypes.java|14|15|93.33%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpWriterJob.java|92|95|96.84%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementEvent.java|363|375|96.80%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java|133|138|96.38%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableStructureAdapter.java|41|42|97.62%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableUpdateDetails.java|92|94|97.87%|
|:large_blue_circle:|io/questdb/std/BoolList.java|91|92|98.91%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpNetworkIOJob.java|51|52|98.08%|
|:large_blue_circle:|io/questdb/std/IntList.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|17|17|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableUtils.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 897 / 924 (97.08%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementEventType.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/DefaultColumnTypes.java|14|15|93.33%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpWriterJob.java|92|95|96.84%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementEvent.java|363|375|96.80%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java|133|138|96.38%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableStructureAdapter.java|41|42|97.62%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableUpdateDetails.java|92|94|97.87%|
|:large_blue_circle:|io/questdb/std/BoolList.java|91|92|98.91%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpNetworkIOJob.java|51|52|98.08%|
|:large_blue_circle:|io/questdb/std/IntList.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|17|17|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableUtils.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 897 / 924 (97.08%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementEventType.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/DefaultColumnTypes.java|14|15|93.33%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpWriterJob.java|92|95|96.84%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementEvent.java|363|375|96.80%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java|133|138|96.38%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableStructureAdapter.java|41|42|97.62%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableUpdateDetails.java|92|94|97.87%|
|:large_blue_circle:|io/questdb/std/BoolList.java|91|92|98.91%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpNetworkIOJob.java|51|52|98.08%|
|:large_blue_circle:|io/questdb/std/IntList.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|17|17|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableUtils.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 897 / 924 (97.08%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementEventType.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/DefaultColumnTypes.java|14|15|93.33%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpWriterJob.java|92|95|96.84%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementEvent.java|363|375|96.80%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java|133|138|96.38%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableStructureAdapter.java|41|42|97.62%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableUpdateDetails.java|92|94|97.87%|
|:large_blue_circle:|io/questdb/std/BoolList.java|91|92|98.91%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpNetworkIOJob.java|51|52|98.08%|
|:large_blue_circle:|io/questdb/std/IntList.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|17|17|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableUtils.java|2|2|100.00%|

File: core/src/main/java/io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java [hunk 1]
```diff
@@ -168,22 +167,22 @@ LineTcpConnectionContext of(long clientFd, IODispatcher<LineTcpConnectionContext
     protected final IOContextResult parseMeasurements(NetworkIOJob netIoJob) {
         while (true) {
             try {
-                ParseResult rc = goodMeasurement ? protoParser.parseMeasurement(recvBufPos) : protoParser.skipMeasurement(recvBufPos);
+                ParseResult rc = goodMeasurement ? parser.parseMeasurement(recvBufPos) : parser.skipMeasurement(recvBufPos);
                 switch (rc) {
                     case MEASUREMENT_COMPLETE: {
                         if (goodMeasurement) {
-                            if (scheduler.tryButCouldNotCommit(netIoJob, protoParser, charSink)) {
+                            if (scheduler.scheduleEvent(netIoJob, parser, floatingDirectCharSink)) {
```
Comment (ideoma) [inline core/src/main/java/io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java:174]: I cannot believe this cryptic name will be done `tryButCouldNotCommit`

**Comment by glasstiger** [comment]
merged in all changes from ilp-fixes
also merged in the conflicting changes from master

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 912 / 947 (96.30%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementEventType.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/std/IntList.java|4|6|66.67%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java|128|138|92.75%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/DefaultColumnTypes.java|14|15|93.33%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpWriterJob.java|96|99|96.97%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementEvent.java|362|374|96.79%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableStructureAdapter.java|41|42|97.62%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableUpdateDetails.java|104|107|97.20%|
|:large_blue_circle:|io/questdb/std/BoolList.java|93|94|98.94%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpNetworkIOJob.java|51|52|98.08%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|17|17|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableUtils.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 912 / 947 (96.30%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementEventType.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/std/IntList.java|4|6|66.67%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java|128|138|92.75%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/DefaultColumnTypes.java|14|15|93.33%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpWriterJob.java|96|99|96.97%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementEvent.java|362|374|96.79%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableStructureAdapter.java|41|42|97.62%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableUpdateDetails.java|104|107|97.20%|
|:large_blue_circle:|io/questdb/std/BoolList.java|93|94|98.94%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpNetworkIOJob.java|51|52|98.08%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|17|17|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableUtils.java|2|2|100.00%|

File: core/src/main/java/io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java (hunk 1 above)
Comment (bluestreak01) [reply core/src/main/java/io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java:174]: me neither!

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 912 / 947 (96.30%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementEventType.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/std/IntList.java|4|6|66.67%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java|128|138|92.75%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/DefaultColumnTypes.java|14|15|93.33%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpWriterJob.java|96|99|96.97%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementEvent.java|362|374|96.79%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableStructureAdapter.java|41|42|97.62%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableUpdateDetails.java|104|107|97.20%|
|:large_blue_circle:|io/questdb/std/BoolList.java|93|94|98.94%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpNetworkIOJob.java|51|52|98.08%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|17|17|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableUtils.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 912 / 947 (96.30%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementEventType.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/std/IntList.java|4|6|66.67%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java|128|138|92.75%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/DefaultColumnTypes.java|14|15|93.33%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpWriterJob.java|96|99|96.97%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementEvent.java|362|374|96.79%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableStructureAdapter.java|41|42|97.62%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableUpdateDetails.java|104|107|97.20%|
|:large_blue_circle:|io/questdb/std/BoolList.java|93|94|98.94%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpNetworkIOJob.java|51|52|98.08%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|17|17|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableUtils.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 912 / 947 (96.30%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementEventType.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/std/IntList.java|4|6|66.67%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java|128|138|92.75%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/DefaultColumnTypes.java|14|15|93.33%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpWriterJob.java|96|99|96.97%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementEvent.java|362|374|96.79%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableStructureAdapter.java|41|42|97.62%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableUpdateDetails.java|104|107|97.20%|
|:large_blue_circle:|io/questdb/std/BoolList.java|93|94|98.94%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpNetworkIOJob.java|51|52|98.08%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|17|17|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableUtils.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 912 / 947 (96.30%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementEventType.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/std/IntList.java|4|6|66.67%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java|128|138|92.75%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/DefaultColumnTypes.java|14|15|93.33%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpWriterJob.java|96|99|96.97%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementEvent.java|362|374|96.79%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableStructureAdapter.java|41|42|97.62%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableUpdateDetails.java|104|107|97.20%|
|:large_blue_circle:|io/questdb/std/BoolList.java|93|94|98.94%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpNetworkIOJob.java|51|52|98.08%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|17|17|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableUtils.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 912 / 947 (96.30%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementEventType.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/std/IntList.java|4|6|66.67%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java|128|138|92.75%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/DefaultColumnTypes.java|14|15|93.33%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpWriterJob.java|96|99|96.97%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementEvent.java|362|374|96.79%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableStructureAdapter.java|41|42|97.62%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableUpdateDetails.java|104|107|97.20%|
|:large_blue_circle:|io/questdb/std/BoolList.java|93|94|98.94%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpNetworkIOJob.java|51|52|98.08%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|17|17|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableUtils.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 912 / 947 (96.30%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementEventType.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/std/IntList.java|4|6|66.67%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java|128|138|92.75%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/DefaultColumnTypes.java|14|15|93.33%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpWriterJob.java|96|99|96.97%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementEvent.java|362|374|96.79%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableStructureAdapter.java|41|42|97.62%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableUpdateDetails.java|104|107|97.20%|
|:large_blue_circle:|io/questdb/std/BoolList.java|93|94|98.94%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpNetworkIOJob.java|51|52|98.08%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|17|17|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableUtils.java|2|2|100.00%|

**Review by glasstiger — APPROVED** [review]

**Review by ideoma — APPROVED** [review]

**Pull request merged**